### ISSUE TITLE
Preset in `CMakeUserPresets.json` using "condition" do not appear in configure preset selection

### ISSUE THREAD

**Issue by theblackunknown** [microsoft/vscode-cmake-tools#2749]
### Brief Issue Summary

When using `CMake: Select Configure preset`, it does not list preset using "condition" feature, whereas similar one - with "condition" - in `CMakePresets.json` are.

exemple of `CMakeUserPresets.json`
```
{
  "version": 3,
  "configurePresets": [
    {
      "name": "foo-visible"
    },
    {
      "name": "foo-non-visible",
      "condition": {"type": "equals", "lhs": "${hostSystemName}", "rhs": "Darwin"}
    }
  ]
}
```

### CMake Tools Diagnostics

```shell
{
  "os": "darwin",
  "vscodeVersion": "1.71.2",
  "cmtVersion": "1.12.27",
  "configurations": [
    {
      "folder": "/Users/andreamachizard/<redacted>",
      "cmakeVersion": "3.24.1",
      "configured": false,
      "generator": "",
      "usesPresets": true,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": null
    }
  ]
}
```
```


### Debug Log

```shell
[presetController] Successfully validated presets in /Users/andreamachizard/devel-adobe/Eclair/CMakePresets.json
[presetController] Successfully validated presets in /Users/andreamachizard/devel-adobe/Eclair/CMakeUserPresets.json
```
```


### Additional Information


Listing in VSCode
<img width="602" alt="image" src="https://user-images.githubusercontent.com/566332/190681187-c63df3ac-d8a9-4777-96a8-6080b3b58898.png">

Listing with `cmake --list-presets`
<img width="387" alt="image" src="https://user-images.githubusercontent.com/566332/190681370-95a4f0f5-9be5-4757-ba81-9dda77d4e23e.png">

**Comment by elahehrashedi** [comment]
Thank you for reporting this issue. We submitted a new PR to address this issue: https://github.com/microsoft/vscode-cmake-tools/pull/2752

**Comment by bobbrow** [comment]
The fix for this will be available in the 1.13.7 pre-release tomorrow morning.
